factorial.rb

```diff
@@ -0,0 +1,15 @@
+def factorial_r(n)
+  if n == 1
+    1
+  else
+    n * factorial_r(n-1)
+  end
+end
+
+def factorial_i(n)
+  fact = 1
+  1.upto(n) do |i|
```

Comment:
#upto, a more elegant solution than I went with. I think I may have used a clunky form of #times